packages/interactive-messages/src/adapter.ts

```diff
@@ -0,0 +1,652 @@
+          }
+        }
+      }
+
+      // if there's no reason to eliminate this callback, then its a match!
+      return true;
+    });
+  }
+}
+
+export default SlackMessageAdapter;
+
+/** Some HTTP response statuses. */
+enum ResponseStatus {
+  Ok = 200,
+  Failure = 500,
+}
+
+/**
+ * The result of a call to {@link SlackMessageAdapter#dispatch}.
+ */
+interface DispatchResult {
+  status: ResponseStatus;
+  content?: any;
+}
+
+/**
+ * Options for constructing {@link SlackMessageAdapter}.
+ */
+export interface MessageAdapterOptions {
+  syncResponseTimeout?: number;
+  lateResponseFallbackEnabled?: boolean;
+}
+
+/**
+ * Constraints on when to call an action handler.
+ */
+export interface ActionConstraints {
+  /**
+   * A string or RegExp to match against the `callback_id`
+   */
+  callbackId?: string | RegExp;
+
+  /**
+   * A string or RegExp to match against the `block_id`
+   */
+  blockId?: string | RegExp;
+
+  /**
+   * A string or RegExp to match against the `action_id`
+   */
+  actionId?: string | RegExp;
+
+  /**
+   * Valid types include all
+   * [actions block elements](https://api.slack.com/reference/messaging/interactive-components),
+   * `select` only for menu selections, or `dialog_submission` only for dialog submissions
+   */
+  type?: string;
+
+  /**
+   * When `true` only match actions from an unfurl
+   */
+  unfurl?: boolean;
+}
+
+/**
+ * Constraints on when to call an options handler.
+ */
+export interface OptionsConstraints {
+  /**
+   * A string or RegExp to match against the `callback_id`
+   */
+  callbackId?: string | RegExp;
+
+  /**
+   * A string or RegExp to match against the `block_id`
+   */
+  blockId?: string | RegExp;
+
+  /**
+   * A string or RegExp to match against the `action_id`
+   */
+  actionId?: string | RegExp;
+
+  /**
+   * The source of options request.
+   */
+  within: 'block_actions' | 'interactive_message' | 'dialog';
+}
+
+type AnyConstraints = ActionConstraints | OptionsConstraints;
+
+/**
+ * The type of stored constraints.
+ */
+const enum StoredConstraintsType {
+  Action = 'action',
+  Options = 'options',
+}
+
+/**
+ * Internal storage type that describes the constraints of an ActionHandler.
+ */
+type StoredConstraints =
+ | ({ handlerType: StoredConstraintsType.Action } & ActionConstraints)
+ | ({ handlerType: StoredConstraintsType.Options } & OptionsConstraints);
+
+/**
+ * A function used to send message updates after an action is handled. This function can be used
+ * up to 5 times in 30 minutes.
+ *
+ * @param message - a [message](https://api.slack.com/docs/interactive-message-field-guide#top-level_message_fields).
+ *   Dialog submissions do not allow `resplace_original: false` on this message. @returnsthere's no contract or
+ *   interface for the resolution value, but this Promise will resolve when the HTTP response from the `response_url`
+ *   request is complete and reject when there is an error.
+ */
+type Respond = (message: any) => Promise<any>;
```

Comment:
This might be better described as returning a `Promise<unknown>`. The idea is that the user calls this function and knows it will complete asynchronously, but knows nothing about the value that the returned Promise resolves to. We actually don't want the user to depend on that value at all (it should be treated as opaque). An `any` would indicate its some meaningful value but the type is anything.